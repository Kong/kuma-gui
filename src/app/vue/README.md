---
section: Packages
---
# vue

Automatic `vue` initialization using our projects dependency injection layer.

