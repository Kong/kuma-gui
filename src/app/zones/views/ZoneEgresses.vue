<template>
  <div class="zoneegresses">
    <FrameSkeleton>
      <DataOverview
        :page-size="pageSize"
        :is-loading="isLoading"
        :error="error"
        :empty-state="empty_state"
        :table-data="tableData"
        :table-data-is-empty="isEmpty"
        :next="next"
        @table-action="tableAction"
        @load-data="loadData($event)"
      >
        <template #additionalControls>
          <KButton
            v-if="$route.query.ns"
            class="back-button"
            appearance="primary"
            :to="{
              name: 'zoneegresses'
            }"
          >
            <span class="custom-control-icon">
              &larr;
            </span>
            View All
          </KButton>
        </template>
      </DataOverview>
      <TabsWidget
        v-if="isEmpty === false"
        :has-error="error !== null"
        :is-loading="isLoading"
        :tabs="tabs"
        initial-tab-override="overview"
      >
        <template #tabHeader>
          <div>
            <h3> Zone Egress: {{ entity.name }}</h3>
          </div>
          <div>
            <EntityURLControl :name="entity.name" />
          </div>
        </template>
        <template #overview>
          <LabelList>
            <div>
              <ul>
                <li
                  v-for="(value, key) in entity"
                  :key="key"
                >
                  <h4 v-if="value">
                    {{ key }}
                  </h4>
                  <p>
                    {{ value }}
                  </p>
                </li>
              </ul>
            </div>
          </LabelList>
        </template>
        <template #insights>
          <KCard border-variant="noBorder">
            <template #body>
              <AccordionList :initially-open="0">
                <AccordionItem
                  v-for="(value, key) in subscriptionsReversed"
                  :key="key"
                >
                  <template #accordion-header>
                    <SubscriptionHeader :details="value" />
                  </template>

                  <template #accordion-content>
                    <SubscriptionDetails
                      :details="value"
                      is-discovery-subscription
                    />
                  </template>
                </AccordionItem>
              </AccordionList>
            </template>
          </KCard>
        </template>

        <template #xds-configuration>
          <EnvoyData
            data-path="xds"
            :zone-egress-name="entity.name"
            query-key="envoy-data-zone-egress"
          />
        </template>

        <template #envoy-stats>
          <EnvoyData
            data-path="stats"
            :zone-egress-name="entity.name"
            query-key="envoy-data-zone-egress"
          />
        </template>

        <template #envoy-clusters>
          <EnvoyData
            data-path="clusters"
            :zone-egress-name="entity.name"
            query-key="envoy-data-zone-egress"
          />
        </template>
      </TabsWidget>
    </FrameSkeleton>
  </div>
</template>

<script>
import { getItemStatusFromInsight } from '@/dataplane'
import { getSome } from '@/helpers'
import { getTableData } from '@/utils/tableDataUtils'
import { PAGE_SIZE_DEFAULT } from '@/consts'
import Kuma from '@/services/kuma'
import AccordionItem from '@/app/common/AccordionItem.vue'
import AccordionList from '@/app/common/AccordionList.vue'
import DataOverview from '@/app/common/DataOverview.vue'
import EntityURLControl from '@/app/common/EntityURLControl.vue'
import EnvoyData from '@/app/common/EnvoyData.vue'
import FrameSkeleton from '@/app/common/FrameSkeleton.vue'
import LabelList from '@/app/common/LabelList.vue'
import SubscriptionDetails from '@/app/common/subscriptions/SubscriptionDetails.vue'
import SubscriptionHeader from '@/app/common/subscriptions/SubscriptionHeader.vue'
import TabsWidget from '@/app/common/TabsWidget.vue'

export default {
  name: 'ZoneEgresses',

  components: {
    AccordionItem,
    AccordionList,
    DataOverview,
    EntityURLControl,
    EnvoyData,
    FrameSkeleton,
    LabelList,
    SubscriptionDetails,
    SubscriptionHeader,
    TabsWidget,
  },

  data() {
    return {
      isLoading: true,
      isEmpty: false,
      error: null,

      empty_state: {
        title: 'No Data',
        message: 'There are no Zone Egresses present.',
      },
      tableData: {
        headers: [
          { key: 'actions', hideLabel: true },
          { label: 'Status', key: 'status' },
          { label: 'Name', key: 'name' },
        ],
        data: [],
      },
      tabs: [
        {
          hash: '#overview',
          title: 'Overview',
        },
        {
          hash: '#insights',
          title: 'Zone Egress Insights',
        },
        {
          hash: '#xds-configuration',
          title: 'XDS Configuration',
        },
        {
          hash: '#envoy-stats',
          title: 'Stats',
        },
        {
          hash: '#envoy-clusters',
          title: 'Clusters',
        },
      ],
      entity: {},
      rawData: [],
      pageSize: PAGE_SIZE_DEFAULT,
      next: null,
      subscriptionsReversed: [],
    }
  },
  watch: {
    $route() {
      // Ensures basic state is reset when switching meshes using the mesh selector.
      this.isLoading = true
      this.isEmpty = false
      this.error = null

      this.init()
    },
  },
  beforeMount() {
    this.init()
  },
  methods: {
    init() {
      this.loadData()
    },
    tableAction(ev) {
      const data = ev

      // load the data into the tabs
      this.getEntity(data)
    },

    async loadData(offset = '0') {
      this.isLoading = true
      this.isEmpty = false

      const query = this.$route.query.ns || null

      try {
        const { data, next } = await getTableData({
          getAllEntities: Kuma.getAllZoneEgressOverviews.bind(Kuma),
          getSingleEntity: Kuma.getZoneEgressOverview.bind(Kuma),
          size: this.pageSize,
          offset,
          query,
        })

        // set pagination
        this.next = next

        // set table data
        if (data.length) {
          this.isEmpty = false
          this.rawData = data
          this.getEntity({ name: data[0].name })

          this.tableData.data = data.map((item) => {
            const { zoneEgressInsight = {} } = item

            return { ...item, ...getItemStatusFromInsight(zoneEgressInsight) }
          })
        } else {
          this.tableData.data = []
          this.isEmpty = true
        }
      } catch (err) {
        if (err instanceof Error) {
          error.value = err
        } else {
          console.error(err)
        }

        this.isEmpty = true
      } finally {
        this.isLoading = false
      }
    },
    getEntity(entity) {
      const selected = ['type', 'name']
      const item = this.rawData.find((data) => data.name === entity.name)

      const subscriptions = item?.zoneEgressInsight?.subscriptions ?? []

      this.subscriptionsReversed = Array.from(subscriptions).reverse()

      this.entity = getSome(item, selected)
    },
  },
}
</script>
