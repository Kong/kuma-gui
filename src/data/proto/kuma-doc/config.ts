/* eslint-disable */

export const protobufPackage = "doc";

export interface Config {
  type: Config_Type;
  name: string;
  fileName: string;
  templateName: string;
}

export enum Config_Type {
  Undefined = "Undefined",
  Policy = "Policy",
  Proxy = "Proxy",
  Other = "Other",
  UNRECOGNIZED = "UNRECOGNIZED",
}

export function config_TypeFromJSON(object: any): Config_Type {
  switch (object) {
    case 0:
    case "Undefined":
      return Config_Type.Undefined;
    case 1:
    case "Policy":
      return Config_Type.Policy;
    case 2:
    case "Proxy":
      return Config_Type.Proxy;
    case 99:
    case "Other":
      return Config_Type.Other;
    case -1:
    case "UNRECOGNIZED":
    default:
      return Config_Type.UNRECOGNIZED;
  }
}

export function config_TypeToJSON(object: Config_Type): string {
  switch (object) {
    case Config_Type.Undefined:
      return "Undefined";
    case Config_Type.Policy:
      return "Policy";
    case Config_Type.Proxy:
      return "Proxy";
    case Config_Type.Other:
      return "Other";
    case Config_Type.UNRECOGNIZED:
    default:
      return "UNRECOGNIZED";
  }
}

function createBaseConfig(): Config {
  return { type: Config_Type.Undefined, name: "", fileName: "", templateName: "" };
}

export const Config = {
  fromJSON(object: any): Config {
    return {
      type: isSet(object.type) ? config_TypeFromJSON(object.type) : Config_Type.Undefined,
      name: isSet(object.name) ? String(object.name) : "",
      fileName: isSet(object.fileName) ? String(object.fileName) : "",
      templateName: isSet(object.templateName) ? String(object.templateName) : "",
    };
  },

  toJSON(message: Config): unknown {
    const obj: any = {};
    message.type !== undefined && (obj.type = config_TypeToJSON(message.type));
    message.name !== undefined && (obj.name = message.name);
    message.fileName !== undefined && (obj.fileName = message.fileName);
    message.templateName !== undefined && (obj.templateName = message.templateName);
    return obj;
  },

  fromPartial(object: DeepPartial<Config>): Config {
    const message = Object.create(createBaseConfig()) as Config;
    message.type = object.type ?? Config_Type.Undefined;
    message.name = object.name ?? "";
    message.fileName = object.fileName ?? "";
    message.templateName = object.templateName ?? "";
    return message;
  },
};

type Builtin = Date | Function | Uint8Array | string | number | boolean | undefined;

export type DeepPartial<T> = T extends Builtin ? T
  : T extends Array<infer U> ? Array<DeepPartial<U>> : T extends ReadonlyArray<infer U> ? ReadonlyArray<DeepPartial<U>>
  : T extends {} ? { [K in keyof T]?: DeepPartial<T[K]> }
  : Partial<T>;

function isSet(value: any): boolean {
  return value !== null && value !== undefined;
}
