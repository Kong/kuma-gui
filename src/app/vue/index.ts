import { createApp } from 'vue'
import {
  createRouter,
  createWebHistory,
} from 'vue-router'

import type { ServiceDefinition } from '@/services/utils'
import { token, createInjections } from '@/services/utils'
import type { Component } from 'vue'
import type { Router, RouteRecordRaw, NavigationGuard } from 'vue-router'

type Token = ReturnType<typeof token>

type VueApp = ReturnType<typeof createApp>
export type PluginDefinition = Parameters<VueApp['use']>
export type ComponentDefinition = [string, Component]

const $ = {
  app: token<(App: Component) => Promise<VueApp>>('vue.app'),
  router: token<Router>('vue.router'),

  components: token('vue.components'),
  plugins: token<PluginDefinition[]>('vue.plugins'),
  routes: token<RouteRecordRaw[]>('vue.routes'),
  routesLabel: token<RouteRecordRaw[]>('vue.routes.label'),
  navigationGuards: token<NavigationGuard[]>('vue.routes.navigation.guards'),
}
export const services = (app: Record<string, Token>): ServiceDefinition[] => {
  return [
    [$.app, {
      service: (
        components: ComponentDefinition[],
        plugins: PluginDefinition[],
      ) => {
        return async (App: Component) => {
          const app = createApp(App)

          plugins.forEach(([...args]) => {
            app.use(...args)
          })

          components.forEach(([name, component]: [string, Component]) => {
            app.component(name, component)
          })

          return app
        }
      },
      arguments: [
        $.components,
        $.plugins,
      ],
    }],
    [$.router, {
      service: (
        env: (str: 'KUMA_BASE_PATH') => string,
        routes: RouteRecordRaw[],
        guards: NavigationGuard[],
      ) => {
        const router = createRouter({
          history: createWebHistory(env('KUMA_BASE_PATH')),
          routes,
        })

        guards.forEach((item) => {
          if (typeof item === 'function') {
            router.beforeEach(item)
          }
        })

        return router
      },
      arguments: [
        app.env,
        $.routes,
        $.navigationGuards,
      ],
    }],

    [token('vue.plugins'), {
      service: (router) => {
        return [
          [router],
        ]
      },
      arguments: [
        $.router,
      ],
      labels: [
        $.plugins,
      ],
    }],

    // TODO(jc): Label decoration. Make it so we can decorate labels
    // Temporarily turn the routesLabel label into routes for decoration
    // purposes
    [$.routes, {
      service: (routes: RouteRecordRaw[]) => routes,
      arguments: [
        $.routesLabel,
      ],
    }],
    //

    // TODO(jc): Make it so we don't need to provide a [] empty default for
    // labels
    [token('application.routes.navigation.guards'), {
      service: () => {
        return []
      },
      labels: [
        $.navigationGuards,
      ],
    }],
  ]
}

export const TOKENS = $
export const [
  useRouter,
] = createInjections(
  TOKENS.router,
)
