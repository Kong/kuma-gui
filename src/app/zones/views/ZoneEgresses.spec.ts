import { flushPromises } from '@vue/test-utils'
import { render, screen } from '@testing-library/vue'
import userEvent from '@testing-library/user-event'

import ZoneEgresses from './ZoneEgresses.vue'
import { store, storeKey } from '@/store/store'
import { ClientConfigInterface } from '@/store/modules/config/config.types'
import * as config from '@/api/mock-data/config.json'

jest.mock('@/utilities/helpers', () => {
  const originalModule = jest.requireActual('@/utilities/helpers')

  return {
    __esModule: true,
    ...originalModule,
    humanReadableDate: (data: string) => data,
    rawReadableDate: (data: string) => data,
  }
})

function renderComponent(mode = 'standalone') {
  const clientConfig: ClientConfigInterface = { ...config, mode }
  store.state.config.clientConfig = clientConfig

  return render(ZoneEgresses, {
    global: {
      plugins: [[store, storeKey]],
    },
  })
}

describe('ZoneEgresses.vue', () => {
  it('renders snapshot when no multizone', async () => {
    const { container } = renderComponent()

    await flushPromises()

    await screen.findByText(/ZoneEgressOverview/)

    expect(container).toMatchSnapshot()
  })

  it('renders snapshot when multizone', async () => {
    const { container } = renderComponent('global')

    await flushPromises()

    await screen.findByText(/ZoneEgressOverview/)

    expect(container).toMatchSnapshot()
  })

  it('renders zoneegress insights', async () => {
    renderComponent('global')

    await screen.findByText(/ZoneEgressOverview/)

    await userEvent.click(screen.getByText(/Zone Egress Insights/))
    expect(screen.getByTestId('tab-container')).toMatchSnapshot()
  })
})
