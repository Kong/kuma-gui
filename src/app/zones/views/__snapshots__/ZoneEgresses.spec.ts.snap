// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ZoneEgresses.vue renders snapshot when multizone 1`] = `
<div
  class="zoneegresses"
>
  <div
    class="kcard-stack"
  >
    <div
      class="kcard-border"
    >
      <div
        class="data-overview"
        data-testid="data-overview"
      >
        <div
          class="data-table-controls mb-2"
        >
          
          
          <button
            class="k-button medium rounded primary refresh-button"
            data-testid="data-overview-refresh-button"
            type="button"
          >
            
            <span
              class="k-button-icon kong-icon kong-icon-redo"
            >
              <svg
                fill="white"
                height="16"
                role="img"
                viewBox="0 0 17 14"
                width="16"
                xmlns="http://www.w3.org/2000/svg"
              >
                
  
                <title>
                  Redo
                </title>
                
  
                <path
                  clip-rule="evenodd"
                  d="M12.466 12.06a6.868 6.868 0 112.196-5.676l.85-.85a.847.847 0 011.201-.017c.327.327.315.869-.017 1.2L14.233 9.18c-.342.342-.88.35-1.208.023L10.654 6.83c-.324-.324-.314-.859.028-1.2.34-.34.881-.348 1.2-.029l.847.846a4.938 4.938 0 10-1.63 4.245l1.367 1.367z"
                  fill="white"
                  fill-rule="evenodd"
                />
                

              </svg>
              

            </span>
            
            
             Refresh 
            
            <!---->
          </button>
        </div>
        <div
          class="data-overview-content"
        >
          <!-- data -->
          <div
            class="data-overview-table"
          >
            <div
              class="k-table-container data-overview-table"
              data-testid="data-overview-table"
              is-clickable=""
            >
              <!---->
              <section
                class="k-table-wrapper"
              >
                <table
                  class="k-table has-hover is-clickable"
                  data-tableid="aaaabbbb-cccc-dddd-eeee-ffffffffffff"
                >
                  <thead
                    class=""
                  >
                    <tr
                      class=""
                    >
                      
                      <th
                        class=""
                      >
                        <span
                          class="d-flex align-items-center"
                        >
                          
                          <span
                            class=""
                          >
                            Status
                          </span>
                          
                          <!---->
                        </span>
                      </th>
                      <th
                        class=""
                      >
                        <span
                          class="d-flex align-items-center"
                        >
                          
                          <span
                            class=""
                          >
                            Name
                          </span>
                          
                          <!---->
                        </span>
                      </th>
                      
                    </tr>
                  </thead>
                  <tbody>
                    
                    <tr
                      class="is-selected"
                      role="link"
                      tabindex="0"
                    >
                      
                      <td
                        class=""
                      >
                        
                        <span
                          class="status status--with-title status--success"
                          data-testid="status-badge"
                        >
                          <span
                            class=""
                          >
                            online
                          </span>
                        </span>
                        
                      </td>
                      <td
                        class=""
                      >
                        
                        
                        zoneegress-1
                        
                        
                      </td>
                      
                    </tr>
                    
                  </tbody>
                </table>
                <!---->
              </section>
            </div>
            <div
              class="pagination"
            >
              <!--v-if-->
              <!--v-if-->
            </div>
          </div>
          <!--v-if-->
          <!--v-if-->
        </div>
      </div>
    </div>
    <div
      class="kcard-border"
    >
      <div
        class="tab-container"
        data-testid="tab-container"
      >
        
        <header
          class="tab__header"
        >
          
          <h1
            class="entity-heading"
          >
             Zone Egress: zoneegress-1
          </h1>
          
        </header>
        <div
          class="tab__content-container"
        >
          <div
            class="k-tabs"
          >
            <ul
              aria-label="Tabs"
              role="tablist"
            >
              
              <li
                aria-controls="panel-0"
                aria-selected="true"
                class="tab-item active"
                id="overview-tab"
                role="tab"
                tabindex="0"
              >
                <a
                  class="tab-link"
                >
                  
                  Overview
                  
                </a>
              </li>
              <li
                aria-controls="panel-1"
                aria-selected="false"
                class="tab-item"
                id="insights-tab"
                role="tab"
                tabindex="0"
              >
                <a
                  class="tab-link"
                >
                  
                  Zone Egress Insights
                  
                </a>
              </li>
              <li
                aria-controls="panel-2"
                aria-selected="false"
                class="tab-item"
                id="xds-configuration-tab"
                role="tab"
                tabindex="0"
              >
                <a
                  class="tab-link"
                >
                  
                  XDS Configuration
                  
                </a>
              </li>
              <li
                aria-controls="panel-3"
                aria-selected="false"
                class="tab-item"
                id="envoy-stats-tab"
                role="tab"
                tabindex="0"
              >
                <a
                  class="tab-link"
                >
                  
                  Stats
                  
                </a>
              </li>
              <li
                aria-controls="panel-4"
                aria-selected="false"
                class="tab-item"
                id="envoy-clusters-tab"
                role="tab"
                tabindex="0"
              >
                <a
                  class="tab-link"
                >
                  
                  Clusters
                  
                </a>
              </li>
              
            </ul>
            
            <div
              aria-labelledby="overview-tab"
              class="tab-container"
              id="panel-0"
              role="tabpanel"
              tabindex="0"
            >
              
              <section
                aria-describedby="aaaabbbb-cccc-dddd-eeee-ffffffffffff"
                class="kong-card noBorder"
              >
                <!---->
                <!---->
                <div
                  class="k-card-content d-flex"
                >
                  <div
                    class="k-card-body"
                    id="aaaabbbb-cccc-dddd-eeee-ffffffffffff"
                  >
                    
                    
                    <div
                      class="label-list"
                    >
                      <div
                        class="label-list__content"
                      >
                        
                        <div>
                          <ul>
                            
                            <li>
                              <h4>
                                type
                              </h4>
                              <p>
                                ZoneEgressOverview
                              </p>
                            </li>
                            <li>
                              <h4>
                                name
                              </h4>
                              <p>
                                zoneegress-1
                              </p>
                            </li>
                            
                          </ul>
                        </div>
                        
                      </div>
                    </div>
                    
                    
                  </div>
                  <!---->
                </div>
              </section>
              
            </div>
            <div
              aria-labelledby="insights-tab"
              class="tab-container"
              id="panel-1"
              role="tabpanel"
              tabindex="0"
            >
              <!---->
            </div>
            <div
              aria-labelledby="xds-configuration-tab"
              class="tab-container"
              id="panel-2"
              role="tabpanel"
              tabindex="0"
            >
              <!---->
            </div>
            <div
              aria-labelledby="envoy-stats-tab"
              class="tab-container"
              id="panel-3"
              role="tabpanel"
              tabindex="0"
            >
              <!---->
            </div>
            <div
              aria-labelledby="envoy-clusters-tab"
              class="tab-container"
              id="panel-4"
              role="tabpanel"
              tabindex="0"
            >
              <!---->
            </div>
            
          </div>
        </div>
        
      </div>
    </div>
  </div>
</div>
`;

exports[`ZoneEgresses.vue renders snapshot when no multizone 1`] = `
<div
  class="zoneegresses"
>
  <div
    class="kcard-stack"
  >
    <div
      class="kcard-border"
    >
      <div
        class="data-overview"
        data-testid="data-overview"
      >
        <div
          class="data-table-controls mb-2"
        >
          
          
          <button
            class="k-button medium rounded primary refresh-button"
            data-testid="data-overview-refresh-button"
            type="button"
          >
            
            <span
              class="k-button-icon kong-icon kong-icon-redo"
            >
              <svg
                fill="white"
                height="16"
                role="img"
                viewBox="0 0 17 14"
                width="16"
                xmlns="http://www.w3.org/2000/svg"
              >
                
  
                <title>
                  Redo
                </title>
                
  
                <path
                  clip-rule="evenodd"
                  d="M12.466 12.06a6.868 6.868 0 112.196-5.676l.85-.85a.847.847 0 011.201-.017c.327.327.315.869-.017 1.2L14.233 9.18c-.342.342-.88.35-1.208.023L10.654 6.83c-.324-.324-.314-.859.028-1.2.34-.34.881-.348 1.2-.029l.847.846a4.938 4.938 0 10-1.63 4.245l1.367 1.367z"
                  fill="white"
                  fill-rule="evenodd"
                />
                

              </svg>
              

            </span>
            
            
             Refresh 
            
            <!---->
          </button>
        </div>
        <div
          class="data-overview-content"
        >
          <!-- data -->
          <div
            class="data-overview-table"
          >
            <div
              class="k-table-container data-overview-table"
              data-testid="data-overview-table"
              is-clickable=""
            >
              <!---->
              <section
                class="k-table-wrapper"
              >
                <table
                  class="k-table has-hover is-clickable"
                  data-tableid="aaaabbbb-cccc-dddd-eeee-ffffffffffff"
                >
                  <thead
                    class=""
                  >
                    <tr
                      class=""
                    >
                      
                      <th
                        class=""
                      >
                        <span
                          class="d-flex align-items-center"
                        >
                          
                          <span
                            class=""
                          >
                            Status
                          </span>
                          
                          <!---->
                        </span>
                      </th>
                      <th
                        class=""
                      >
                        <span
                          class="d-flex align-items-center"
                        >
                          
                          <span
                            class=""
                          >
                            Name
                          </span>
                          
                          <!---->
                        </span>
                      </th>
                      
                    </tr>
                  </thead>
                  <tbody>
                    
                    <tr
                      class="is-selected"
                      role="link"
                      tabindex="0"
                    >
                      
                      <td
                        class=""
                      >
                        
                        <span
                          class="status status--with-title status--success"
                          data-testid="status-badge"
                        >
                          <span
                            class=""
                          >
                            online
                          </span>
                        </span>
                        
                      </td>
                      <td
                        class=""
                      >
                        
                        
                        zoneegress-1
                        
                        
                      </td>
                      
                    </tr>
                    
                  </tbody>
                </table>
                <!---->
              </section>
            </div>
            <div
              class="pagination"
            >
              <!--v-if-->
              <!--v-if-->
            </div>
          </div>
          <!--v-if-->
          <!--v-if-->
        </div>
      </div>
    </div>
    <div
      class="kcard-border"
    >
      <div
        class="tab-container"
        data-testid="tab-container"
      >
        
        <header
          class="tab__header"
        >
          
          <h1
            class="entity-heading"
          >
             Zone Egress: zoneegress-1
          </h1>
          
        </header>
        <div
          class="tab__content-container"
        >
          <div
            class="k-tabs"
          >
            <ul
              aria-label="Tabs"
              role="tablist"
            >
              
              <li
                aria-controls="panel-0"
                aria-selected="true"
                class="tab-item active"
                id="overview-tab"
                role="tab"
                tabindex="0"
              >
                <a
                  class="tab-link"
                >
                  
                  Overview
                  
                </a>
              </li>
              <li
                aria-controls="panel-1"
                aria-selected="false"
                class="tab-item"
                id="insights-tab"
                role="tab"
                tabindex="0"
              >
                <a
                  class="tab-link"
                >
                  
                  Zone Egress Insights
                  
                </a>
              </li>
              <li
                aria-controls="panel-2"
                aria-selected="false"
                class="tab-item"
                id="xds-configuration-tab"
                role="tab"
                tabindex="0"
              >
                <a
                  class="tab-link"
                >
                  
                  XDS Configuration
                  
                </a>
              </li>
              <li
                aria-controls="panel-3"
                aria-selected="false"
                class="tab-item"
                id="envoy-stats-tab"
                role="tab"
                tabindex="0"
              >
                <a
                  class="tab-link"
                >
                  
                  Stats
                  
                </a>
              </li>
              <li
                aria-controls="panel-4"
                aria-selected="false"
                class="tab-item"
                id="envoy-clusters-tab"
                role="tab"
                tabindex="0"
              >
                <a
                  class="tab-link"
                >
                  
                  Clusters
                  
                </a>
              </li>
              
            </ul>
            
            <div
              aria-labelledby="overview-tab"
              class="tab-container"
              id="panel-0"
              role="tabpanel"
              tabindex="0"
            >
              
              <section
                aria-describedby="aaaabbbb-cccc-dddd-eeee-ffffffffffff"
                class="kong-card noBorder"
              >
                <!---->
                <!---->
                <div
                  class="k-card-content d-flex"
                >
                  <div
                    class="k-card-body"
                    id="aaaabbbb-cccc-dddd-eeee-ffffffffffff"
                  >
                    
                    
                    <div
                      class="label-list"
                    >
                      <div
                        class="label-list__content"
                      >
                        
                        <div>
                          <ul>
                            
                            <li>
                              <h4>
                                type
                              </h4>
                              <p>
                                ZoneEgressOverview
                              </p>
                            </li>
                            <li>
                              <h4>
                                name
                              </h4>
                              <p>
                                zoneegress-1
                              </p>
                            </li>
                            
                          </ul>
                        </div>
                        
                      </div>
                    </div>
                    
                    
                  </div>
                  <!---->
                </div>
              </section>
              
            </div>
            <div
              aria-labelledby="insights-tab"
              class="tab-container"
              id="panel-1"
              role="tabpanel"
              tabindex="0"
            >
              <!---->
            </div>
            <div
              aria-labelledby="xds-configuration-tab"
              class="tab-container"
              id="panel-2"
              role="tabpanel"
              tabindex="0"
            >
              <!---->
            </div>
            <div
              aria-labelledby="envoy-stats-tab"
              class="tab-container"
              id="panel-3"
              role="tabpanel"
              tabindex="0"
            >
              <!---->
            </div>
            <div
              aria-labelledby="envoy-clusters-tab"
              class="tab-container"
              id="panel-4"
              role="tabpanel"
              tabindex="0"
            >
              <!---->
            </div>
            
          </div>
        </div>
        
      </div>
    </div>
  </div>
</div>
`;

exports[`ZoneEgresses.vue renders zoneegress insights 1`] = `
<div
  class="tab-container"
  data-testid="tab-container"
>
  
  <header
    class="tab__header"
  >
    
    <h1
      class="entity-heading"
    >
       Zone Egress: zoneegress-1
    </h1>
    
  </header>
  <div
    class="tab__content-container"
  >
    <div
      class="k-tabs"
    >
      <ul
        aria-label="Tabs"
        role="tablist"
      >
        
        <li
          aria-controls="panel-0"
          aria-selected="false"
          class="tab-item"
          id="overview-tab"
          role="tab"
          tabindex="0"
        >
          <a
            class="tab-link"
          >
            
            Overview
            
          </a>
        </li>
        <li
          aria-controls="panel-1"
          aria-selected="true"
          class="tab-item active"
          id="insights-tab"
          role="tab"
          tabindex="0"
        >
          <a
            class="tab-link"
          >
            
            Zone Egress Insights
            
          </a>
        </li>
        <li
          aria-controls="panel-2"
          aria-selected="false"
          class="tab-item"
          id="xds-configuration-tab"
          role="tab"
          tabindex="0"
        >
          <a
            class="tab-link"
          >
            
            XDS Configuration
            
          </a>
        </li>
        <li
          aria-controls="panel-3"
          aria-selected="false"
          class="tab-item"
          id="envoy-stats-tab"
          role="tab"
          tabindex="0"
        >
          <a
            class="tab-link"
          >
            
            Stats
            
          </a>
        </li>
        <li
          aria-controls="panel-4"
          aria-selected="false"
          class="tab-item"
          id="envoy-clusters-tab"
          role="tab"
          tabindex="0"
        >
          <a
            class="tab-link"
          >
            
            Clusters
            
          </a>
        </li>
        
      </ul>
      
      <div
        aria-labelledby="overview-tab"
        class="tab-container"
        id="panel-0"
        role="tabpanel"
        tabindex="0"
      >
        <!---->
      </div>
      <div
        aria-labelledby="insights-tab"
        class="tab-container"
        id="panel-1"
        role="tabpanel"
        tabindex="0"
      >
        
        <section
          aria-describedby="aaaabbbb-cccc-dddd-eeee-ffffffffffff"
          class="kong-card noBorder"
        >
          <!---->
          <!---->
          <div
            class="k-card-content d-flex"
          >
            <div
              class="k-card-body"
              id="aaaabbbb-cccc-dddd-eeee-ffffffffffff"
            >
              
              
              <ul
                class="accordion-list"
              >
                
                
                <li
                  class="accordion-item active"
                >
                  <button
                    aria-expanded="true"
                    class="accordion-item-header"
                    data-testid="accordion-item-button"
                    type="button"
                  >
                    
                    <h4
                      class="text-lg font-medium"
                    >
                      <span
                        class="color-green-500"
                      >
                         Connect time: July 13, 2021 at 8:41:04 AM
                      </span>
                      <!--v-if-->
                    </h4>
                    
                  </button>
                  <transition-stub
                    appear="false"
                    css="true"
                    name="accordion"
                    persisted="false"
                  >
                    <div
                      class="accordion-item-content"
                      data-testid="accordion-item-content"
                    >
                      
                      <div>
                        <div>
                          <h5
                            class="overview-tertiary-title"
                          >
                             General Information: 
                          </h5>
                          <ul>
                            <!--v-if-->
                            <li>
                              <strong>
                                Control Plane Instance ID:
                              </strong>
                              <span>
                                kuma-control-plane-84f5589874-nmspq-0867
                              </span>
                            </li>
                            <li>
                              <strong>
                                Last Connected:
                              </strong>
                               July 13, 2021
                            </li>
                            <!--v-if-->
                          </ul>
                        </div>
                        <div>
                          <ul
                            class="overview-stat-grid"
                          >
                            
                            <li>
                              <h6
                                class="overview-tertiary-title"
                              >
                                cds: 
                              </h6>
                              <ul>
                                
                                <li>
                                  <strong>
                                    Responses Sent:
                                  </strong>
                                  <span>
                                    3
                                  </span>
                                </li>
                                <li>
                                  <strong>
                                    Responses Acknowledged:
                                  </strong>
                                  <span>
                                    3
                                  </span>
                                </li>
                                
                              </ul>
                            </li>
                            <li>
                              <h6
                                class="overview-tertiary-title"
                              >
                                eds: 
                              </h6>
                              <ul>
                                
                                <li>
                                  <strong>
                                    Responses Sent:
                                  </strong>
                                  <span>
                                    2
                                  </span>
                                </li>
                                <li>
                                  <strong>
                                    Responses Acknowledged:
                                  </strong>
                                  <span>
                                    3
                                  </span>
                                </li>
                                
                              </ul>
                            </li>
                            <li>
                              <h6
                                class="overview-tertiary-title"
                              >
                                lds: 
                              </h6>
                              <ul>
                                
                                <li>
                                  <strong>
                                    Responses Sent:
                                  </strong>
                                  <span>
                                    3
                                  </span>
                                </li>
                                <li>
                                  <strong>
                                    Responses Acknowledged:
                                  </strong>
                                  <span>
                                    3
                                  </span>
                                </li>
                                
                              </ul>
                            </li>
                            <li>
                              <h6
                                class="overview-tertiary-title"
                              >
                                rds: 
                              </h6>
                              <ul>
                                
                                
                              </ul>
                            </li>
                            
                          </ul>
                        </div>
                      </div>
                      
                    </div>
                  </transition-stub>
                </li>
                
                
              </ul>
              
              
            </div>
            <!---->
          </div>
        </section>
        
      </div>
      <div
        aria-labelledby="xds-configuration-tab"
        class="tab-container"
        id="panel-2"
        role="tabpanel"
        tabindex="0"
      >
        <!---->
      </div>
      <div
        aria-labelledby="envoy-stats-tab"
        class="tab-container"
        id="panel-3"
        role="tabpanel"
        tabindex="0"
      >
        <!---->
      </div>
      <div
        aria-labelledby="envoy-clusters-tab"
        class="tab-container"
        id="panel-4"
        role="tabpanel"
        tabindex="0"
      >
        <!---->
      </div>
      
    </div>
  </div>
  
</div>
`;
